src/Utilities/Math.hpp

```diff
@@ -65,6 +67,57 @@ constexpr T step_function(const T& arg) noexcept {
   return static_cast<T>((arg >= static_cast<T>(0)) ? 1 : 0);
 }
 
+/*!
+ * \ingroup UtilitiesGroup
+ * \brief Smoothly interpolates from 0 to 1 between `left_edge` and `right_edge`
+ * with a Hermite polynomial of degree `2 * Order + 1`.
+ *
+ * The smoothstep function is
+ *
+ * \f{align*}
+ * S_n(x) = \begin{cases}
+ * 0 &\quad \text{for} \quad x\leq x_0 \\
+ * \tilde{S}_n((x - x_0) / (x_1 - x_0))
+ * &\quad \text{for} \quad x_0 \leq x\leq x_1 \\
+ * 1 &\quad \text{for} \quad x_1\leq x \\
+ * \end{cases}
+ * \f}
+ *
+ * where \f$x_0\f$ is `left_edge`, \f$x_1\f$ is `right_edge`, and, up to third
+ * order,
+ *
+ * \f{align*}
+ * \tilde{S}_0(x) &= x \\
+ * \tilde{S}_1(x) &= 3x^2 - 2x^3 \\
+ * \tilde{S}_2(x) &= 10x^3 - 15x^4 + 6x^5 \\
+ * \tilde{S}_3(x) &= 35x^4 - 84x^5 + 70x^6 - 20x^7
+ * \text{.}
+ * \f}
+ */
+template <size_t Order, typename DataType>
+DataType smoothstep(const DataType& arg, const double left_edge,
```

Comment:
What do you mean with different interface here? If you are referring to the additional `left_edge` and `right_edge` arguments: The step function has no width, but a smoothstep does. It's convenient to be able to specify the `left_edge` and `right_edge` when calling this function, instead of having to figure out how to rescale the `arg`. I could default `left_edge=0.` and `right_edge=1.` if you think that could be useful.

Reply:
Would you ever have the transition not occur about `x=0.5`? It seems like specifying a `transition_width` or `smoothing_width` would be more intuitive.

Reply:
Both `x=0.5` and `x=0` are reasonable choices for the transition, in fact the smoothstep approaches the step function for `N->\infty` at `x=0`, not `x=0.5`. Specifying both bounds avoids the ambiguity, so I think that's preferable. The implementations in OpenGL (https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/smoothstep.xhtml) and by Nvidia (https://developer.download.nvidia.com/cg/smoothstep.html) also specify the bounds.

Reply:
Of course, they also have a different ordering of the arguments, so it seems that, given your argument we should follow what they're doing and have `lower_edge, upper_edge, x` as the arguments. Sound okay?

Reply:
sure, done